docs: native sidecar containers support (#1485)
kyma-project/istio: docs/user/00-00-istio-sidecar-proxies.md

The Istio module does not restart an Istio sidecar proxy if it has a custom image set. See [Resource Annotations](https://istio.io/latest/docs/reference/config/annotations/#SidecarProxyImage).

> There are two types of sidecar containers - regular ones and Kubernetes native sidecars. Istio Module supports restart of both types.

Reviewer: ```suggestion
The Istio module supports restarting both types of sidecar containers: regular ones and Kubernetes native sidecars.
```
